linode/manager
# upcoming: [M3-7429] - Add Parent/Proxy 'Switch Account' button and drawer to user profile dropdown menu

## Description 📝
This PR introduces a more intuitive design for the profile dropdown menu and implements a new drawer specifically for listing child accounts. The profile dropdown menu should now visually distinguish between parent and proxy account types, providing a clearer representation of the user's account hierarchy.

> [!NOTE]  
> This story focuses solely on enhancing the user interface and does not involve token issuance or the actual switching to the proxy account, which will be handled separately (in M3-7430).

## Changes  🔄
Subtle differences between the two drawers are bolded.

**Parent View:**
- **We should see which account the parent user is logged into (parent company).**
- "Switch Account" button should be visible.
- "Switch Account" button should open "Switch Account" drawer with list of child accounts.
- The child accounts should be scrollable if the list is too long.
   - For the purposes of this PR, we just expect the drawer to scroll; a follow up PR will look into a virtualized list. 
- Upon closing drawer, profile dropdown menu should remain open.

**Proxy View:**
- **We should see which company the parent user is logged into (child company).**
- "Switch Account" button should be visible.
- "Switch Account" button should open "Switch Account" drawer with list of child accounts.
- **There should be a link to switch back to parent account.**
- Upon closing drawer, profile dropdown menu should remain open.

**Child View:**
- No change, same as typical user without a parent/child relationship

## Preview 📷
<details>
<summary>Screenshots</summary>

| Before  | After   | 
| ------- | ------- |
| User Menu Popover | ![Screenshot 2024-01-05 at 4 25 02 PM](https://github.com/linode/manager/assets/114685994/8ea02ed5-9c8c-46e3-afcf-5b3f62d72734) ![Screenshot 2024-01-05 at 4 24 55 PM](https://github.com/linode/manager/assets/114685994/56aabb4c-cef4-42be-b81b-f8c098f79ef3) |
| Switch Account Drawer - Parent View | ![Screenshot 2024-01-05 at 4 10 32 PM](https://github.com/linode/manager/assets/114685994/53ce9e33-a6d1-4f75-a0e8-8a52aa6db7f0) |
| Switch Account Drawer - Proxy View |![Screenshot 2024-01-05 at 4 11 03 PM](https://github.com/linode/manager/assets/114685994/8e94863a-d217-4908-a2d2-092dd7d84534) |
| Switch Account Drawer - Error Views | ![Screenshot 2024-01-05 at 4 11 57 PM](https://github.com/linode/manager/assets/114685994/74935099-4476-424c-9172-9e12d2d61322) ![Screenshot 2024-01-05 at 4 11 30 PM](https://github.com/linode/manager/assets/114685994/c45d03c3-01c0-4df8-ab56-9a0b194158b7)|
</details>

## How to test 🧪

### Prerequisites
(How to setup test environment)

- Check out this PR and `yarn dev`.
- Make sure the Parent Child Account Access feature flag is on and mocks are on.

### Verification steps 
(How to verify changes)

1. Click on the Profile & Account button in the top menu to open the user menu popover.
2. Observe that the `Mock Company` name is displayed and a Switch Account button is visible.
3. Confirm the changes described above in "Parent View". If you want to test a long list of child accounts, uncomment the return statement on [L1180](https://github.com/linode/manager/pull/10031/files#diff-c4904f695ece84849bc2372518313ff64475b5598ab35b45708266fa74e2e6e8R1180) of the mocks.
4. Go to `serverHandlers.ts` and edit the `account/users/:user` request to the following in order to mock a proxy user:
```
  rest.get('*/account/users/:user', (req, res, ctx) => {
  // Parent/Child: switch the `user_type` depending on what account view you need to mock.
  return res(ctx.json(accountUserFactory.build({ user_type: 'proxy' })));
  }),
```
5. Repeat steps 1-3 above for proxy view.
6. Repeat step 4 for a Child user and confirm parity with prod.
7. Toggle off feature flag and confirm parity with prod.
8. Verify unit tests pass:
```
yarn test UserMenu SwitchAccountDrawer

```
 
## As an Author I have considered 🤔

*Check all that apply*
